comtypes/_memberspec.py

```diff
@@ -175,54 +175,65 @@ def call_with_inout(self, *args, **kw):
         # Indexed by order in the output
         outargs = {}
         outnum = 0
+        param_index = 0
+        # Go through all expected arguments and match them to the provided arguments.
+        # param_index first counts through the positional and then through the keyword arguments.
         for i, info in enumerate(paramflags):
             direction = info[0]
-            if direction & 3 == 3:
+            dir_in = direction & 1 == 1
+            dir_out = direction & 2 == 2
+            is_positional = param_index < len(args)
+            if dir_in and dir_out:
                 # This is an [in, out] parameter.
                 #
                 # Determine name and required type of the parameter.
                 name = info[1]
                 # [in, out] parameters are passed as pointers,
                 # this is the pointed-to type:
-                atyp = argtypes[i]._type_
+                atyp: type[_CData] = getattr(argtypes[i], "_type_")
```

Comment:
I think that this line is not covered by the current test.

Because Py <= 3.8, the `type` was not ["Type Hinting Generics In Standard Collections"](https://peps.python.org/pep-0585/), which should result in an error if this line is executed.

Reply:
I was very surprised to find that this line is indeed covered by the test, but it still does not crash in Python 3.7. In any case, I will fix it.